solr/core/src/java/org/apache/solr/update/processor/ParseDateFieldUpdateProcessorFactory.java

```diff
@@ -159,8 +170,9 @@ public void init(NamedList args) {
     Collection<String> formatsParam = args.removeConfigArgs(FORMATS_PARAM);
     if (null != formatsParam) {
       for (String value : formatsParam) {
-        DateTimeFormatter formatter = new DateTimeFormatterBuilder().parseCaseInsensitive()
-            .appendPattern(value).toFormatter(locale).withZone(defaultTimeZone);
+        DateTimeFormatter formatter = new DateTimeFormatterBuilder().parseLenient().parseCaseInsensitive()
```

Comment:
I see you're hard-coding leniency and not making it an option.  It's fine, I think.  I'm not sure why someone would expressly want it to be strict (would want an error), not for an application of parsing like we see here.

Reply:
I guess it can be added as an option,
should I work on it?

Reply:
nah; it's fine.  Less to document :-)

Reply:
Less work is always welcome 👍